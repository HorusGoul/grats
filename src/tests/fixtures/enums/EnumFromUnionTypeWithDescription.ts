/** @gqlType */
class Query {
  /** @gqlField */
  hello: MyEnum;
}

/**
 * Hello!
 * @gqlEnum
 */
type MyEnum = "VALID" | "INVALID";
