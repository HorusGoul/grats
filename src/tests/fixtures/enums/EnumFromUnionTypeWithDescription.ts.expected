-----------------
INPUT
----------------- 
/** @gqlType */
class SomeType {
  /** @gqlField */
  hello: MyEnum;
}

/**
 * Hello!
 * @gqlEnum
 */
type MyEnum = "VALID" | "INVALID";

-----------------
OUTPUT
-----------------
directive @methodName(name: String!) on FIELD_DEFINITION

directive @exported(filename: String!, functionName: String!) on FIELD_DEFINITION

type SomeType {
  hello: MyEnum
}

"""Hello!"""
enum MyEnum {
  VALID
  INVALID
}